scripts/metrics/metrics_xml_to_csv.py

```diff
@@ -0,0 +1,187 @@
+"""
+Copyright (c) Contributors to the Open 3D Engine Project.
+For complete copyright and license terms please see the LICENSE at the root of this distribution.
+
+SPDX-License-Identifier: Apache-2.0 OR MIT
+
+Scrapes metrics from xml files and creates csv formatted files.
+"""
+import os.path
+import sys
+import csv
+import argparse
+import xml.etree.ElementTree as xmlElementTree
+import datetime
+
+from common import logging
+
+DEFAULT_CTEST_LOG_FILENAME = 'Test.xml'
+TAG_FILE = 'TAG'
+TESTING_DIR = 'Testing'
+
+def now_wrap():
+    """
+    Wrapper function for now() for unit testing purposes.
+    @return: datetime.now()
+    """
+    return datetime.datetime.now()
+
+
+def _get_default_csv_filename():
+    # Format default file name based off of date
+    now = now_wrap()
+    return f"{now.year}_{now.month:02d}_{now.day:02d}.csv"
+
+
+# Setup logging.
+logger = logging.get_logger("test_metrics")
+logging.setup_logger(logger)
+
+# Create the csv field header
+CTEST_FIELDS_HEADER = [
+    'test_name',
+    'status',
+    'duration_seconds'
+]
+
+
+def main():
+    # Parse args
+    args = parse_args()
+
+    # Load global configuration.
+    logger.info("Loading configuration")
+
+    # Construct the full path to the xml file
+    file_path = _get_test_xml_path(args.build_folder, args.ctest_log)
+
+    # Create csv files
+    with open(args.csv_file, 'w', encoding='UTF8', newline='') as csv_file:
```

Comment:
`'w'` will overwrite a file that already exists. Should this instead emit an error if a CSV file already existed at that location? May be appropriate to warn the user that the file got overwritten.